src/components/Proposals/ProposalDetails/VoteActions.js

```diff
@@ -0,0 +1,220 @@
+              action.
+            </Info>
+          </>
+        )}
+      </>
+    )
+  }
+
+  if (canUserVote && hasVoted && !changeVote) {
+    return (
+      <div>
+        <Button
+          mode="strong"
+          onClick={handleChangeVote}
+          wide
+          css={`
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          Change my vote
+        </Button>
+        <Info>
+          While the voting period is open, you can{' '}
+          <strong>change your vote</strong> as many times as you wish.
+        </Info>
+      </div>
+    )
+  }
+
+  if (canUserVote) {
+    return (
+      <div>
+        {connectedAccount ? (
+          <>
+            <Buttons onClickYes={onVoteYes} onClickNo={onVoteNo} />
+            <TokenReference
+              snapshotBlock={snapshotBlock}
+              startDate={startDate}
+              tokenSymbol={tokenSymbol}
+              userBalance={userBalance}
+              userBalanceNow={userBalanceNow}
+            />
+          </>
+        ) : (
+          <div
+            css={`
+              border-radius: ${RADIUS}px;
+              background: ${theme.background};
+              padding: ${3.5 * GU}px ${10 * GU}px;
+              text-align: center;
+            `}
+          >
+            <div
+              css={`
+                ${textStyle('body1')};
+              `}
+            >
+              You must enable your account to vote on this proposal
+            </div>
+            <div
+              css={`
+                ${textStyle('body2')};
+                color: ${theme.surfaceContentSecondary};
+                margin-top: ${2 * GU}px;
+              `}
+            >
+              Connect to your Ethereum provider by clicking on the{' '}
+              <strong
+                css={`
+                  display: inline-flex;
+                  align-items: center;
+                  position: relative;
+                  top: 7px;
+                `}
+              >
+                <IconConnect /> Enable account
+              </strong>{' '}
+              button on the header. You may be temporarily redirected to a new
+              screen.
+            </div>
+          </div>
+        )}
+      </div>
+    )
+  }
+
+  return (
+    <div>
+      <Buttons disabled />
+      <Info mode="warning">
+        {userBalanceNow > 0
+          ? 'Although the currently connected account holds tokens, it'
+          : 'The currently connected account'}{' '}
+        did not hold any <strong>{tokenSymbol}</strong> tokens when this vote
+        began ({dateFormat(toMs(startDate))}) and therefore cannot participate
+        in this vote. Make sure your accounts are holding{' '}
+        <strong>{tokenSymbol}</strong> at the time a vote begins if you'd like
+        to vote using this Voting app.
+      </Info>
+    </div>
+  )
+})
+
+const Buttons = ({ onClickYes = noop, onClickNo = noop, disabled = false }) => (
```

Comment:
We can remove default noops if they are provided as defaultProps to the component :)